modules/lang-mustache/src/test/java/org/elasticsearch/script/mustache/SearchTemplateRequestXContentTests.java

```diff
@@ -0,0 +1,197 @@
+                .field("profile", false)
+            .endObject();
+
+        XContentBuilder actualRequest = XContentFactory.contentBuilder(contentType);
+        request.toXContent(actualRequest, ToXContent.EMPTY_PARAMS);
+
+        assertToXContentEquivalent(
+            BytesReference.bytes(expectedRequest),
+            BytesReference.bytes(actualRequest),
+            contentType);
+    }
+
+    public void testFromXContentWithEmbeddedTemplate() throws Exception {
+        String source = "{" +
+                "    'source' : {\n" +
+                "    'query': {\n" +
+                "      'terms': {\n" +
+                "        'status': [\n" +
+                "          '{{#status}}',\n" +
+                "          '{{.}}',\n" +
+                "          '{{/status}}'\n" +
+                "        ]\n" +
+                "      }\n" +
+                "    }\n" +
+                "  }" +
+                "}";
+
+        SearchTemplateRequest request = SearchTemplateRequest.fromXContent(newParser(source));
+        assertThat(request.getScript(), equalTo("{\"query\":{\"terms\":{\"status\":[\"{{#status}}\",\"{{.}}\",\"{{/status}}\"]}}}"));
+        assertThat(request.getScriptType(), equalTo(ScriptType.INLINE));
+        assertThat(request.getScriptParams(), nullValue());
+    }
+
+    public void testFromXContentWithEmbeddedTemplateAndParams() throws Exception {
+        String source = "{" +
+            "    'source' : {" +
+            "      'query': { 'match' : { '{{my_field}}' : '{{my_value}}' } }," +
+            "      'size' : '{{my_size}}'" +
+            "    }," +
+            "    'params' : {" +
+            "        'my_field' : 'foo'," +
+            "        'my_value' : 'bar'," +
+            "        'my_size' : 5" +
+            "    }" +
+            "}";
+
+        SearchTemplateRequest request = SearchTemplateRequest.fromXContent(newParser(source));
+        assertThat(request.getScript(), equalTo("{\"query\":{\"match\":{\"{{my_field}}\":\"{{my_value}}\"}},\"size\":\"{{my_size}}\"}"));
+        assertThat(request.getScriptType(), equalTo(ScriptType.INLINE));
+        assertThat(request.getScriptParams().size(), equalTo(3));
+        assertThat(request.getScriptParams(), hasEntry("my_field", "foo"));
+        assertThat(request.getScriptParams(), hasEntry("my_value", "bar"));
+        assertThat(request.getScriptParams(), hasEntry("my_size", 5));
+    }
+
+    public void testFromXContentWithMalformedRequest() {
+        // Unclosed template id
+        expectThrows(XContentParseException.class, () -> SearchTemplateRequest.fromXContent(newParser("{'id' : 'another_temp }")));
+    }
+
+    /**
+     * Creates a {@link XContentParser} with the given String while replacing single quote to double quotes.
+     */
+    private XContentParser newParser(String s) throws IOException {
+        assertNotNull(s);
+        return createParser(JsonXContent.jsonXContent, s.replace("'", "\""));
```

Comment:
you have this for readability of the snippets above? Or some other reason too?

Reply:
This set-up comes from some pre-existing tests that I decided not to rewrite -- I think it's simply for readability.